diff --git a/lxd/instance/drivers/driver_qemu.go b/lxd/instance/drivers/driver_qemu.go
index 77804f670..cffc252eb 100644
--- a/lxd/instance/drivers/driver_qemu.go
+++ b/lxd/instance/drivers/driver_qemu.go
@@ -3836,9 +3836,10 @@ func (d *qemu) addNetDevConfig(busName string, qemuDev map[string]string, bootIn
 				return nil, fmt.Errorf("Error creating new ifreq for %q: %w", nicName, err)
 			}
 
-			// These settings need to match those set by the LXD device that created the interface.
-			ifr.SetUint16(unix.IFF_TAP | unix.IFF_NO_PI | unix.IFF_MULTI_QUEUE)
-
+			// These settings need to be compatible with what the LXD device created the interface with
+			// and what QEMU is expecting.
+			ifr.SetUint16(unix.IFF_TAP | unix.IFF_NO_PI | unix.IFF_ONE_QUEUE | unix.IFF_MULTI_QUEUE | unix.IFF_VNET_HDR)
+      
 			// Sets the file handle to point to the requested NIC interface.
 			err = unix.IoctlIfreq(int(f.Fd()), unix.TUNSETIFF, ifr)
 			if err != nil {
